src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/AbstractConditionMapper.java

```diff
@@ -101,26 +135,67 @@ private void validateUpdatedFieldIsNotInPrimaryKey(String updatedField, String p
         }
     }
 
-    private String getUpdateSetClause(RequestWrapper request, String field, AttributeValue value) {
+    protected void validateUpdatedFieldIsNotInIndexKey(String updatedField, DynamoSecondaryIndex index) {
+        PrimaryKey primaryKey = index.getPrimaryKey();
+        if (updatedField.equals(primaryKey.getHashKey()) || (primaryKey.getRangeKey().isPresent()
+            && updatedField.equals(primaryKey.getRangeKey().get()))) {
+            throw new IllegalArgumentException(
+                String.format("Cannot update attribute %s. This secondary index is part of the index key",
```

Comment:
Nitpick: would be better if error message specifies that this is for ADD only. e.g., "Cannot update attribute $s with ADD"